builder/common/run_config.go

```diff
@@ -843,6 +851,13 @@ func (c *RunConfig) Prepare(ctx *interpolate.Context) []error {
 		}
 	}
 
+	if c.SpotAllocationStrategy != "" {
+		if !slices.Contains(ec2.SpotAllocationStrategy_Values(), c.SpotAllocationStrategy) {
```

Comment:
I believe both `if`s can be mutualised
```suggestion
		if c.SpotAllocationStrategy != "" && !slices.Contains(ec2.SpotAllocationStrategy_Values(), c.SpotAllocationStrategy) {
```

Reply:
Also to reply to your question in a comment from a previous review: using `slices.Contains` is fine by my standards! No need to make it manual, I suspect those were made this way in other cases because either the function did not exist at the time it was added, or whomever implemented it wasn't aware it existed, functionally it is essentially the same, might as well use what the library offers us.

Reply:
Addressed it.